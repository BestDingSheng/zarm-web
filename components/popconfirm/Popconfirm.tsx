import React, { Component } from 'react';
import Popper from 'zarm/lib/popper';
import Button from '../button';
import PropsType from './PropsType';
import Icon from '../Icon';
import LocaleReceiver from '../locale-provider/LocaleReceiver';

const noop = () => { };

class Popconfirm extends Component<PropsType, any> {
  static defaultProps = {
    prefixCls: 'zw-popconfirm',
    className: null,
    trigger: 'click',
    direction: 'bottomLeft',
    onCancel: noop,
    icon: <Icon size="sm" type="warning-round" theme="warning" />,
    onOk: noop,
    content: null,
    hasArrow: true,
  };

  private popperRef: React.RefObject<any> = React.createRef();


  constructor(props) {
    super(props);
    this.state = {};
  }


  private handleClose = (): void => {
    this.popperRef.current.handleClose();
  };

  private handleCancel = (): void => {
    const { onCancel } = this.props;
    onCancel();
    this.handleClose();
  };

  private handleConfirm = (): void => {
    const { onOk } = this.props;
    onOk();
    this.handleClose();
  };

  render() {
    const { children, content, cancelText, okText, locale, icon, ...others } = this.props;
    const { prefixCls } = this.props;

    const popContent: React.ReactNode = (
      <>
        <div className={`${prefixCls}__message`}>
          {icon}
          <div className={`${prefixCls}__message--title`}>{content}</div>
        </div>
        <div className={`${prefixCls}__buttons`}>
          <Button
            size="sm"
            onClick={this.handleCancel}
          >
            {cancelText || locale!.cancel}
          </Button>
          <Button size="sm" theme="primary" onClick={this.handleConfirm}>
            {okText || locale!.confirm}
          </Button>
        </div>
      </>
    );

    return (
      <Popper
        ref={this.popperRef}
        content={popContent}
        {...others}
      >
        {children}
      </Popper>
    );
  }
}

export default LocaleReceiver('Popconfirm')(Popconfirm);
