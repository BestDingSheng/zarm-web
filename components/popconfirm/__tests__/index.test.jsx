import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Popconfirm from '../index';
import Button from '../../button';

describe('PopConfirm', () => {
  it('renders PopConfirm correctly', () => {
    const wrapper = render(
      <div>
        <Popconfirm content="确认删除吗" direction="top">
          <Button>Delete</Button>
        </Popconfirm>
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
