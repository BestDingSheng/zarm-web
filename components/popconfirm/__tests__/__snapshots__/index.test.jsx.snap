// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopConfirm renders PopConfirm correctly 1`] = `
<div>
  <div
    class="ui-popover"
  >
    <div
      class="ui-popover-content ui-popover-content-radius"
    >
      <div>
        确认删除吗
        <div
          class="ui-popover-actions"
        >
          <button
            class="za-button za-button--default za-button--radius"
            type="button"
          >
            取消
          </button>
          <button
            class="za-button za-button--primary za-button--radius"
            type="button"
          >
            确定
          </button>
        </div>
      </div>
      <span
        class="ui-popover-arrow"
      />
    </div>
    <button
      class="za-button za-button--default za-button--radius"
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
